src/Deserializers/EntityDeserializer.php

```diff
@@ -39,38 +38,29 @@ public function __construct(
 	 * @throws DeserializationException
 	 */
 	public function deserialize( $serialization ) {
-		if ( !is_array( $serialization ) ) {
-			throw new DeserializationException( 'Entity serialization must be an array' );
-		}
-
 		if ( $this->currentDeserializer->isDeserializerFor( $serialization ) ) {
 			return $this->currentDeserializer->deserialize( $serialization );
-		} elseif ( $this->isLegacySerialization( $serialization ) ) {
+		} elseif ( $this->legacyDeserializer->isDeserializerFor( $serialization ) ) {
 			return $this->legacyDeserializer->deserialize( $serialization );
-		} else {
-			return $this->fromUnknownSerialization( $serialization );
 		}
-	}
 
-	private function isLegacySerialization( array $serialization ) {
-		// This element is called 'id' in the current serialization.
-		return array_key_exists( 'entity', $serialization );
+		throw new DeserializationException(
+			'The provided entity serialization is neither legacy nor current'
+		);
 	}
```

Comment:
If I remember correctly it was done like this rather than using the `DispatchableDeserializer` route because this approach was much cheaper. I did not verify this is indeed the case.